test/goaway_test.go

```diff
@@ -761,3 +761,60 @@ func (s) TestTwoGoAwayPingFrames(t *testing.T) {
 		t.Fatalf("Error waiting for graceful shutdown of the server: %v", err)
 	}
 }
+
+func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
+	// Set up a buffer connection listener
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Failed to listen: %v", err)
+	}
+	defer lis.Close()
+
+	// Create your gRPC server
+	s := grpc.NewServer()
+	defer s.Stop()
+
+	// Start serving on the buffer listener
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+
+	// Connect to the server (using the dialer for our buffer)
+	conn, err := net.DialTimeout("tcp", lis.Addr().String(), defaultTestTimeout)
+	if err != nil {
+		t.Fatalf("Error dialing: %v", err)
+	}
+	defer conn.Close()
+
+	// Create a helper to read frames (similar to your serverTester)
+	st := newServerTesterFromConn(t, conn)
+
+	// Channel to signal if the GOAWAY frame was received
```

Comment:
Prefer using complete sentences. i.e., punctuate the comment like an actual sentence (like terminating with a period).

Reply:
See. https://google.github.io/styleguide/go/decisions#doc-comments